### DISCUSSION TITLE
Size mode for Controls to not strech windows when using AutoSized windows

### DISCUSSION THREAD

**Discussion by sn4k3** [AvaloniaUI/Avalonia#5906] [Q&A] [answered]
In some of my windows i'm using `SizeToContent = SizeToContent.WidthAndHeight;` to accommodate dynamic UserControls
Problem starts when user inputs are stretchable. Example:

![UVtools_2021-05-11_17-05-29](https://user-images.githubusercontent.com/113281/117848365-24059800-b27b-11eb-8396-71ada61064d0.png)

![UVtools_2021-05-11_17-05-37](https://user-images.githubusercontent.com/113281/117848399-2c5dd300-b27b-11eb-82d4-5aed26fee1e5.png)

I want to prevent that stretching issue, obvious i can do via code behind, but is there any property i can set to prevent this?
Would be nice to have a `WidthSizeMode` and `HeightSizeMode` which allows to: `Constrain`, `Auto`

`Spawn` = Auto until control is shown and fixed to it bounds after it
`Constrain` = As current is but don't go beyond parent width/height
`Auto` = As current is

**Comment by maxkatz6** [comment, marked as answer]
Right now there is expected behavior with SizeToContent. Window content is not width-constrained by anything, as well as your visual containers, that's why your TextBox is not limited.

About your options, I am pretty sure developer can write his own Panel with these properties and override Measure correctly.

**Reply by sn4k3** [reply]
Thanks, i will try your hint and try to implement them in a panel
